lib/3scale/api/http_client.rb

```diff
@@ -5,19 +5,22 @@
 module ThreeScale
   module API
     class HttpClient
-      attr_reader :endpoint, :admin_domain, :provider_key, :headers, :format
+      attr_reader :endpoint, :admin_domain, :provider_key, :access_token, :headers, :format
 
-      def initialize(endpoint:, provider_key:, format: :json)
+      def initialize(endpoint:, provider_key: nil, access_token: nil, format: :json)
```

Comment:
Guess there is no reason to leak the provider/access_token concept there. They are treated the same so they can be passed in `authentication`.